python/hsfs/engine/spark.py

```diff
@@ -350,6 +350,16 @@ def write_training_dataset(
             dataset = dataset.read()
 
         dataset = self.convert_to_default_dataframe(dataset)
+
+        # apply transformation functions
+        for (
+            feature_name,
+            transformation_fn,
+        ) in training_dataset.transformation_functions.items():
+            dataset = dataset.withColumn(
+                feature_name, transformation_fn.transformation_fn(col(feature_name))
+            )
```

Comment:
Just FYI, `withColumn` is going to break if you have lots of transformations.

Reply:
`transformation_fn(col(feature_name)).alias(feature_name) )` this will be more efficient?  I understand that `withColumn` will create new dataframe every time its called since dataframe is immutable. But not sure if `alias` can avoid that

Reply:
It's ok, it was just a warning. We can fix it if it becomes a problem.